iosHackthonApp/Modules/Features/Login/Sources/Login/LoginViewModel.swift

```diff
@@ -10,37 +10,66 @@ import SwiftUI
 import shared
 import Components
 import Theming
+import Strings
 
 public final class LoginViewModel: ObservableObject {
 
-	@Published var email: String = ""
-	@Published var password: String = ""
-	@Published var showingAlert = false
-	@Published var isLoading: Bool = false
-	@Published var coloredNavAppearance = UINavigationBarAppearance()
-	@Published var firebase: FirebaseAuthenticationStore
-	@ObservedObject var authorizationStore: AuthorizationStore
-
-	public init(firebase: FirebaseAuthenticationStore, authorizationStore: AuthorizationStore) {
-		self.firebase = firebase
-		self.authorizationStore = authorizationStore
-		coloredNavAppearance.configureWithOpaqueBackground()
-		coloredNavAppearance.backgroundColor = UIColor(ColorManager.appPrimary)
-		coloredNavAppearance.titleTextAttributes = [.foregroundColor: UIColor.white]
-		coloredNavAppearance.largeTitleTextAttributes = [.foregroundColor: UIColor.white]
-	}
-
-	func login(email: String, password: String) {
-		//TODO: API key to be handled in shared layer. remove the apikey parameter once implemented
-		// To run the project for now, add the api key and run
-		isLoading = true
-		firebase.signIn(apiKey: "", email: email, password: password, returnSecureToken: true, completionHandler: { result, error in
-			if (result?.idToken != nil) {
-				self.authorizationStore.storeUserLoggedInStatus(true)
-			} else {
-				self.showingAlert.toggle()
-			}
-			self.isLoading = false
-		})
-	}
+    typealias strings = Strings.Login
+
+    @Published var firebase: FirebaseAuthenticationStore
+    @ObservedObject var authorizationStore: AuthorizationStore
+
+    public init(firebase: FirebaseAuthenticationStore, authorizationStore: AuthorizationStore) {
+        self.firebase = firebase
+        self.authorizationStore = authorizationStore
+        coloredNavAppearance.configureWithOpaqueBackground()
+        coloredNavAppearance.backgroundColor = UIColor(ColorManager.appPrimary)
+        coloredNavAppearance.titleTextAttributes = [.foregroundColor: UIColor.white]
+        coloredNavAppearance.largeTitleTextAttributes = [.foregroundColor: UIColor.white]
+    }
+
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var showingAlert = false
+    @Published var isLoading: Bool = false
+    @Published var coloredNavAppearance = UINavigationBarAppearance()
+
+    let title = strings.heading
+    let emailTitle = strings.email
+    let emailPlaceholder = strings.emailPlaceholder
+    let passwordTitle = strings.password
+    let passwordPlaceholder = strings.passwordPlaceholder
+    let loginButtonTitle = strings.loginButtonTitle
+    let signupButtonTitle = strings.signupButtonTitle
+
+    let alertTitle = strings.invalidLoginTitle
+    let alertMessage = strings.invalidLoginMessage
+    let alertButton = Strings.Common.ok
+
+    let textFieldStyle = GreyTextFieldStyle()
```

Comment:
These are the real changes to this file. Just moving the strings in and the textfieldStyle.
I rearranged the order too so that properties that need to be initialised are above the init, and everything else below, and added typealias for the Strings